tsdb/head.go

```diff
@@ -1597,7 +1602,7 @@ func (s *stripeSeries) gc(mint int64) (map[uint64]struct{}, int) {
 				// series alike.
 				// If we don't hold them all, there's a very small chance that a series receives
 				// samples again while we are half-way into deleting it.
-				j := int(series.ref & stripeMask)
+				j := int(series.ref) & (s.size - 1)
```

Comment:
`s.size - 1` as the mask assumes that the size is a multiple of 2. To avoid any additional logic to take care of non-multiples of 2, we can document it in `NewHead`.

Reply:
Added a comment in NewHead. Do you think we should also include a logic guard in NewHead or just leave it as a comment?

Reply:
Comment should be enough